### PR TITLE
Fix compiler complaints

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

**Comment by avsej** [comment]
@daverigby, @snej ping

**Comment by borrrden** [comment]
@snej I never noticed this PR.  @avsej What platform is giving these complaints?

**Comment by avsej** [comment]
@borrrden on Fedora 26, 27. `gcc (GCC) 7.2.1 20170915 (Red Hat 7.2.1-2)`

**Comment by borrrden** [comment]
We aren't planning on supporting the GCC compiler, so the caveat here will be "as long as it does not interfere with clang."  I'll find some time to see if these changes have any negative effects.

**Comment by avsej** [comment]
so these changes are clang specific? enum member name really looks like used incorrectly

**Comment by avsej** [comment]
> We aren't planning on supporting the GCC compiler

screenshot_2018-01-24_17-39-52

@borrrden, could you fix the README in this case, that the project does not support GCC?

File: Fleece/FleeceCpp.hh [hunk 1]
```diff
@@ -249,7 +249,7 @@ namespace fleeceapi {
 
         AllocedDict(const AllocedDict &d)
         :Dict(d)
-        ,alloc_slice(d)
+        ,alloc_slice(static_cast<const fleece::alloc_slice&>(d))
```
Comment (snej) [inline Fleece/FleeceCpp.hh]: Could you change this to simply `d.data()`?

**Comment by snej** [comment]
Thanks for the patch. I'd like to support GCC in Fleece, since it's a widely-useable component.

**Comment by KraemerDEM** [comment]
I think this can be closed as compiling with gcc nowadays works without this patch

**Pull request closed without merging**